spylib/webhook.py

```diff
@@ -18,3 +23,104 @@ class WebhookCreate(Enum):
+            'privateMetafieldNamespaces': private_metafield_namespaces,
+            'callbackUrl': callback_url,
+        },
+    }
+    res = await offline_token.execute_gql(
+        query=WEBHOOK_CREATE_GQL,
+        operation_name=WebhookCreate.HTTP.value,
+        variables=variables,
+    )
+    webhook_create = res.get(WebhookCreate.HTTP.value, None)
+    if webhook_create and webhook_create.get('userErrors', None):
+        raise ShopifyGQLUserError(res)
+    return WebhookResponse(id=webhook_create['webhookSubscription']['id'])
+
+
+async def create_event_bridge(
+    offline_token: OfflineTokenABC,
+    topic: Union[WebhookTopic, str],
+    arn: str,
+    include_fields: Optional[List[str]] = None,
+    metafield_namespaces: Optional[List[str]] = None,
+    private_metafield_namespaces: Optional[List[str]] = None,
+) -> WebhookResponse:
+    """Uses graphql to subscribe to a webhook and associated it with
+    and AWS Event Bridge with ARN (Amazon Resource Name)"""
+    variables = {
+        'topic': topic.value if isinstance(topic, WebhookTopic) else topic,
+        'webhookSubscription': {
+            'format': 'JSON',
+            'includeFields': include_fields,
+            'metafieldNamespaces': metafield_namespaces,
+            'privateMetafieldNamespaces': private_metafield_namespaces,
+            'arn': arn,
+        },
+    }
+    res = await offline_token.execute_gql(
+        query=WEBHOOK_CREATE_GQL,
+        operation_name=WebhookCreate.EVENT_BRIDGE.value,
+        variables=variables,
+    )
+    webhook_create = res.get(WebhookCreate.EVENT_BRIDGE.value, None)
+    if webhook_create and webhook_create.get('userErrors', None):
+        raise ShopifyGQLUserError(res)
+    return WebhookResponse(id=webhook_create['webhookSubscription']['id'])
+
+
+async def create_pub_sub(
+    offline_token: OfflineTokenABC,
+    topic: Union[WebhookTopic, str],
+    pub_sub_project: str,
+    pub_sub_topic: str,
+    include_fields: Optional[List[str]] = None,
+    metafield_namespaces: Optional[List[str]] = None,
+    private_metafield_namespaces: Optional[List[str]] = None,
+) -> WebhookResponse:
+    """Uses graphql to subscribe to a webhook and associate it with a Google PubSub endpoint"""
+    variables = {
+        'topic': topic.value if isinstance(topic, WebhookTopic) else topic,
+        'webhookSubscription': {
+            'format': 'JSON',
+            'includeFields': include_fields,
+            'metafieldNamespaces': metafield_namespaces,
+            'privateMetafieldNamespaces': private_metafield_namespaces,
+            'pubSubProject': pub_sub_project,
+            'pubSubTopic': pub_sub_topic,
+        },
+    }
+    res = await offline_token.execute_gql(
+        query=WEBHOOK_CREATE_GQL,
+        operation_name=WebhookCreate.PUB_SUB.value,
+        variables=variables,
+    )
+    webhook_create = res.get(WebhookCreate.PUB_SUB.value, None)
+    if webhook_create and webhook_create.get('userErrors', None):
```

Comment:
Maybe you have a reason to be explicit ... `None` is the default return for get if the field doesn't exist.

```suggestion
    webhook_create = res.get(WebhookCreate.PUB_SUB.value)
    if webhook_create and webhook_create.get('userErrors'):
```

Reply:
cool didn't know that